### PR TITLE
Add slug generation service

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Description of changes:*

- Add slug generation
- Implement unique slug

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by VineetBala-AOT — APPROVED** [review]

File: met-api/src/met_api/services/engagement_slug_service.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            raise ValueError(f'No engagement slug found for {slug}')
+        return engagement_slug
+
+    @classmethod
+    def create_engagement_slug(cls, engagement_id: int) -> EngagementSlugModel:
+        """Create an engagement slug."""
+        existing_slug = EngagementSlugModel.find_by_engagement_id(engagement_id)
+        if existing_slug:
+            raise ValueError(f'Engagement slug already exists for engagement {engagement_id}')
+
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/engagement_slug_service.py:25]: may be dont throw error ;just dont generate
slug generation shouldnt affect the actual functionality from where it gets invoked

File: met-api/src/met_api/services/slug_generation_service.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    @staticmethod
+    def generate_slug(text: str) -> str:
+        """Generate a slug from the provided text."""
+        normalized_text = slugify(
+            text,
+            to_lower=True,  # NOSONAR # to_lower is a valid paramter for awesome-slugify
+            max_length=100
+        )
+        return normalized_text
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/slug_generation_service.py:16]: write some texts for this function..
just write a fixture with expected and return value
like Broadway Subway Project to broadway-subway etc
Also max length can be a configuration item.. 
default it to 20 or something

**Review by saravanpa-aot — APPROVED** [review]

File: met-api/src/met_api/services/slug_generation_service.py (hunk 2 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/services/slug_generation_service.py:16]: Added a test with fixtures and made the max length a config item. Not very confident about defaulting to 20, I feel it's too short. I kept it 100 unless 20 is a strong feeling

File: met-api/src/met_api/services/engagement_slug_service.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        suffix_separator = '-'
+        suffix_numbers = []
+        for similar_slug in similar_slugs:
+            slug_parts = similar_slug.slug.split(normalized_slug + suffix_separator)
+            if len(slug_parts) > 1:
+                suffix = slug_parts[-1]
+                if suffix.isdigit():
+                    suffix_numbers.append(int(suffix))
+
+        counter = max(suffix_numbers) + 1 if suffix_numbers else 1
```
[the remark this answers is not in the document]
Comment (jadmsaadaot) [reply met-api/src/met_api/services/engagement_slug_service.py:51]: done, change will be reflected in the next PR

**Pull request merged**